src/components/ActivityCollector/createClickActivityCollector.js

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import createEvent from "../DataCollector/createEvent";
+
+const createClickHandler = (collect, lifecycle) => {
+  return clickEvent => {
+    // TODO: Consider safeguarding from the same object being clicked multiple times in rapid succession?
+    const clickedObj = clickEvent.target;
+    const event = createEvent();
+    lifecycle.onClick(event, clickedObj).then(() => {
+      if (!event.isEmpty()) {
```

Comment:
It feels weird to check `isEmpty` when what you want to accomplish is to see if the event should be sent.  I'm wondering if there is a use case for sending nothing (knowing the context would update it).  Conversely, I'm wondering if there is a use case where a component could augment the event data from the clickedObject, but use a different component to determine if the event should be sent.  Does it make sense to have a `shouldSend` and `isShouldSend` methods?  Conversely, the isEmpty works for Personalization and ActivityCollector so maybe I'm overthinking this.

Reply:
Right, so the component creating the event in this scenario will need to know if it should be sent or not. In the current case if it has been modified it needs to get sent. Going forward this might not be sufficient to determine whether to send it or not. If that is the case we can move the responsibility to any logic modifying the event. We could change the naming of the function.